python/ray/air/util/data_batch_conversion.py

```diff
@@ -109,6 +109,58 @@ def convert_pandas_to_batch_type(
         )
 
 
+def _convert_batch_type_to_numpy(
+    data: DataBatchType,
+) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+    """Convert the provided data to a NumPy ndarray or dict of ndarrays.
+
+    Args:
+        data: Data of type DataBatchType
+
+    Returns:
+        A numpy representation of the input data.
+    """
+    from ray.data.extensions.tensor_extension import ArrowTensorType, TensorDtype
+
+    if isinstance(data, np.ndarray):
+        return data
+    elif isinstance(data, dict):
+        if data.keys() == {TENSOR_COLUMN_NAME} and isinstance(
+            data[TENSOR_COLUMN_NAME], np.ndarray
+        ):
+            return data[TENSOR_COLUMN_NAME]
+        else:
+            return data
+    elif pyarrow is not None and isinstance(data, pyarrow.Table):
+        if data.column_names == [TENSOR_COLUMN_NAME] and isinstance(
+            data.schema.types[0], ArrowTensorType
+        ):
+            # If representing a tensor dataset, return as a single numpy array.
+            return data[0].to_numpy()
+        else:
+            output_dict = {}
+            for col_name in data.column_names:
+                output_dict[col_name] = data[col_name].to_numpy()
+            return output_dict
+    elif isinstance(data, pd.DataFrame):
+        if list(data.columns) == [TENSOR_COLUMN_NAME] and isinstance(
+            next(iter(data.dtypes)), TensorDtype
+        ):
+            # If representing a tensor dataset, return as a single numpy array.
+            return data.iloc[:, 0].to_numpy()
+        else:
+            # Else return as a dict of numpy arrays.
+            output_dict = {}
+            for column_name in data:
+                output_dict[column_name] = data[column_name].to_numpy()
+            return output_dict
```

Comment:
for this clause, lets just directly call `convert_pandas_to_batch_type(type=DataType.NUMPY)`? Instead of reimplementing the same logic

Reply:
i see your point but i think it's a bit different for the new path, we consider TENSOR_COLUMN_NAME here but only checks single column in convert_pandas_to_batch_type.

Reply:
we should have consistency across both regardless right

Reply:
i would vote we bring the `TENSOR_COLUMN_NAME` check here to convert_pandas_to_batch_type, thoughts ?

Reply:
ah ... this changes everything again. Looks like we had more implicit assumptions in our tests regarding how we handled single column casting to numpy. Applying this part to replace convert_pandas_to_batch_type lead to unit test failures